scripts/cell_simulation.py

```diff
@@ -146,92 +146,28 @@ def gen_sim_data(
     ############################################################################
     # Parameters
     ############################################################################
-    out_dir = params.out_dir
-    n_cells = params.n_cells
-    n_mutations = params.n_mutations
-    alpha = params.alpha
-    beta = params.beta
-    na_rate = params.na_rate
-    observe_homozygous = params.observe_homozygous
-    strategy = params.strategy
-    verbose = params.verbose
+    params_dict = vars(params)
 
     ############################################################################
-    # Random Seeds
+    # Generate Data
     ############################################################################
-    rng_tree, rng_cell_attachment, rng_noise = random.split(rng, 3)
-
-    ##############################################################################
-    # Generate Trees
-    ##############################################################################
-    #  generate random trees (uniform sampling) as adjacency matrix
-    tree = tree_inf.generate_random_tree(rng_tree, n_nodes=n_mutations)
-
-    ##############################################################################
-    # Attach Cells To Tree
-    ###############################################################################
-    # convert adjacency matrix to self-connected tree - in tree_inference format
-    np.fill_diagonal(tree, 1)
-    # define strategy
-    strategy = tree_inf.CellAttachmentStrategy[strategy]
-    # attach cells to tree - generate perfect mutation matrix
-    perfect_mutation_mat = tree_inf.attach_cells_to_tree(
-        rng_cell_attachment, tree, n_cells, strategy
-    )
-
-    ###############################################################################
-    # Add Noise
-    ################################################################################
-    # add noise to perfect mutation matrix
-    noisy_mutation_mat = None
-    if (beta > 0) or (alpha > 0) or (na_rate > 0):
-        noisy_mutation_mat = tree_inf.add_noise_to_perfect_matrix(
-            rng_noise, perfect_mutation_mat, alpha, beta, na_rate, observe_homozygous
-        )
+    data = tree_inf.gen_sim_data(params_dict, rng)
 
     ################################################################################
     # Save Simulation Results
     ################################################################################
     # make save name and path from parameters
     filename = compose_save_name(params, tree_no=tree_no) + ".json"
-    fullpath = os.path.join(out_dir, filename)
+    fullpath = os.path.join(params.out_dir, filename)
```

Comment:
Nothing to change in this PR, but in the future I'd suggest to take a look at [pathlib](https://docs.python.org/3/library/pathlib.html). Perhaps you'll find it more convenient than `os` and `os.path`.

Reply:
Ah, you already know it! I saw that you used it in another file. Sorry for noise!

Reply:
Yes, you mentioned it before. I should replace the other usages of `os` with `pathlib` in the future.